src/lib/structures/expr.ml

```diff
@@ -2322,6 +2357,8 @@ module Triggers = struct
     e.ty == Ty.Tbool &&
     match e.f with Sy.Form _ -> false | _ -> true
 
+  (* Keep only patterns in [full_trs] whose all the free term
+     variables are free in [f] with the same type. *)
```

Comment:
What does "free with the same type" means?

Reply:
This comment describe the code as it is. I know that comments are supposed to describe why we do something, not how we do it but here I have no idea why we check the type of variables. Variables of the module `Var` have unique identifiers, so `Var.equal x y` should imply that `x` and `y` have the same type in any expressions of `Expr`. I believe we can replace this portion of code by:
```ocaml
  let trs_in_scope full_trs f =
    STRS.filter
      (fun (e, _, _) ->
         Var.Map.for_all
           (fun v _ -> Var.Map.mem v)
           )
           e.vars
      )full_trs
```

Reply:
I put an assert in a separated PR and I run `Marvin` on it.